sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/providers/BigQueryStorageWriteApiSchemaTransformProvider.java

```diff
@@ -0,0 +1,361 @@
+            invalidConfigMessage
+                + "Invalid write disposition was specified. Available dispositions are: ",
+            WRITE_DISPOSITIONS.keySet());
+      }
+
+      // validate schema
+      if (!Strings.isNullOrEmpty(this.getJsonSchema())) {
+        // check if a TableSchema can be deserialized from the input schema string
+        checkNotNull(BigQueryHelpers.fromJsonString(this.getJsonSchema(), TableSchema.class));
+      } else if (this.getUseBeamSchema() == null || !this.getUseBeamSchema()) {
+        // if no schema is provided, create disposition CREATE_NEVER has to be specified.
+        checkArgument(
+            createDisposition == CreateDisposition.CREATE_NEVER,
+            invalidConfigMessage
+                + "Create disposition is CREATE_IF_NEEDED, but no schema was provided.");
+      }
+
+      // validation checks for streaming writes
+      checkArgument(
+          this.getNumStorageWriteApiStreams() == null || this.getNumStorageWriteApiStreams() > 0,
+          invalidConfigMessage + "When set, numStorageWriteApiStreams must be > 0, but was: %s",
+          this.getNumStorageWriteApiStreams());
+      checkArgument(
+          this.getNumFileShards() == null || this.getNumFileShards() > 0,
+          invalidConfigMessage + "When set, numFileShards must be > 0, but was: %s",
+          this.getNumFileShards());
+      checkArgument(
+          this.getTriggeringFrequencySeconds() == null || this.getTriggeringFrequencySeconds() > 0,
+          invalidConfigMessage + "When set, the trigger frequency must be > 0, but was: %s",
+          this.getTriggeringFrequencySeconds());
+    }
+
+    /**
+     * Instantiates a {@link BigQueryStorageWriteApiSchemaTransformConfiguration.Builder} instance.
+     */
+    public static Builder builder() {
+      return new AutoValue_BigQueryStorageWriteApiSchemaTransformProvider_BigQueryStorageWriteApiSchemaTransformConfiguration
+          .Builder();
+    }
+
+    public abstract String getOutputTable();
+
+    @Nullable
+    public abstract String getJsonSchema();
+
+    @Nullable
+    public abstract Boolean getUseBeamSchema();
+
+    @Nullable
+    public abstract String getCreateDisposition();
+
+    @Nullable
+    public abstract String getWriteDisposition();
+
+    @Nullable
+    public abstract Integer getNumStorageWriteApiStreams();
+
+    @Nullable
+    public abstract Integer getNumFileShards();
+
+    @Nullable
+    public abstract Integer getTriggeringFrequencySeconds();
+
+    @Nullable
+    public abstract Boolean getUseAtLeastOnceSemantics();
+
+    /** Builder for {@link BigQueryStorageWriteApiSchemaTransformConfiguration}. */
+    @AutoValue.Builder
+    public abstract static class Builder {
+      public abstract Builder setOutputTable(String tableSpec);
+
+      public abstract Builder setJsonSchema(String jsonSchema);
+
+      public abstract Builder setUseBeamSchema(Boolean useBeamSchema);
```

Comment:
I think we should always use the schema from the input, so we can just ignore these.